### ISSUE TITLE
Crowdin for translations

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy.github.com#130]
We should look at https://crowdin.com/ for translations. NumPy is going to use it for their new website. I think they have a free tier for open source projects.

**Comment by certik** [comment]
Great idea. (The first flag they show is Czech, how did they know?)

**Comment by goanpeca** [comment]
We also use Crowdin for [Spyder](https://crowdin.com/project/spyder) and [JupyterLab](https://crowdin.com/project/jupyterlab)

I would say go for it :-) !

**Comment by asmeurer** [comment]
I think we just need someone to set it up. We already have the translation infrastructure set up for this repo, so it shouldn't be hard. I think it should be fine to have it automatically commit translations to the repo.

**Comment by goanpeca** [comment]
>I think we just need someone to set it up.

Do you have a bot account for this so that PRs are not committed by whoever sets the integration (but the bot) ?

**Comment by asmeurer** [comment]
Yes, we can use @sympy-bot. Although that raises the question, how are the translators attributed in the repo?

**Comment by goanpeca** [comment]
>Although that raises the question, how are the translators attributed in the repo?

The list is on crowdin, maybe we should be copying that information in the repo.... hmmm with a bot :-p

**Comment by goanpeca** [comment]
@pearu?

**Comment by comradekingu** [comment]
@asmeurer If you read the terms and conditions you will notice it is not. The plan requires sub-licensing strings into the closed translation memory of Crowdin. For other damning details, see https://github.com/jitsi/jitsi-meet/issues/5056#issuecomment-612239691
Use Weblate instead, which actually _is_ and has a gratis plan for libre software projects :)

**Comment by asmeurer** [comment]
@comradekingu can you refer to which part of the terms and conditions you are talking about? I don't see it as an issue if crowdin gets a license to use the translations done for the SymPy website, since they would already be liberally licensed. We don't actually have a license file on this repo, but I would say it should be licensed BSD, the same as SymPy itself. So as long as their terms allow us to do that, I don't see a problem. 

I don't have a strong opinion myself as to which service we use, and if someone wants to take the time to set one, that's the one we should use. I only speak one language myself, so I wouldn't participate in the translations. I've heard from people that they like the crowdin interface, so that is why I suggested it. If weblate or some other service seems better, we can set that up instead.

**Comment by comradekingu** [comment]
@asmeurer 
It is impossible to get any overview with the Crowdin interface, and the context to what string is what, and where it enters the fray is lost. That is the interface you get, with no alternative ones to pick from. This is why translations on Crowdin are not consistent, bar maybe very short ones. It is also much harder to get to the source strings from there. Etc. etc.

It is underhanded to pretend sublicensing "supports" libre software. As the case may be, there isn't any license for this repo, but in the case of "numphy" the distinction (depending what you think about the text) is with this part of the 3-clause BSD license: (which would otherwise need to be upheld)

"    * Redistributions of source code must retain the above copyright
       notice, this list of conditions and the following disclaimer."

think what you will about just taking BSD code that way without contributing back.

In any event the term "Crowdin for Open Source" "We support the open source community" libre software package turns out to be only for "non-profit" projects, meaning no [donations](https://www.sympy.org/en/donate.html), no nothing. This even undermines libre software as a concept, by creating confusion as to what libre software is. In the very least it isn't how to grow one. Crowdin itself neither is libre software, nor keeps a translation memory that is. Instead it employs every shady practice under the sun to sell and mine data. Not OK.

<details>
> **web beacons on all users**.

Here are some direct clippings of all other relevant parts of
https://support.crowdin.com/privacy-policy/.
(My comments only to be found betwixt those, everything else verbatim, in the order it appears, minus irrelevant sections. I have made it easier by cutting it down in bold, which doesn't change the content, presented as is).

---

> **We** may **disclose information to third parties** if you **consent** to us doing so, as well as in the following circumstances:

(OK, what constitutes consent, and what does that entail. Can the service be used _at all_ without it?)

> **We work with third party service providers who provide** website, application development, hosting, maintenance, and **other services** for us. These third parties may **have access to, or process Personal Data or Client Data** as part of providing those services for us.
> We may **make certain automatically-collected, aggregated, or otherwise non-personally-identifiable information available to third parties for various purposes, including** (i) compliance with various reporting obligations; (ii) **for business or marketing purposes; or (iii) to assist such parties in understanding our Clients’, Users’ and Visitors’ interests, habits, and usage patterns for** certain **programs, content, services, and/or functionality available through the Service**

> **We** may **disclose Personal Data or other information** if required to do so by law or i**n the good-faith belief that such action is necessary to comply with applicable laws**, in response to a facially valid court order, judicial or other government subpoena or warrant, **or** to **otherwise cooperate with** law enforcement or **other governmental agencies**.

> **We also** reserve the right to **disclose Personal Data or other information that we believe**, in good faith, **is appropriate** or necessary to (i) take precautions against liability, (ii) protect ourselves or others from fraudulent, abusive, or unlawful uses or activity, (iii) investigate and defend ourselves against any third-party claims or allegations, (iv) protect the security or integrity of the Service and any facilities or equipment used to make the Service available, or (v) protect our property or other legal rights, enforce our contracts, or protect the rights, property, or **safety of others**.

(Wait for it)

> **Information about Users and Visitors, including Personal Data**, may be **disclosed and otherwise transferred to an acquirer, successor or assignee** as part of any merger, acquisition, debt financing, sale of assets, or similar transaction, as well as in the event of an insolvency, bankruptcy, or **receivership in which information is transferred to one or more third parties as one of our business assets** and only if the recipient of the User or Visitor Data commits to a Privacy Policy that has terms substantially consistent with this Privacy Policy.

> **Client Data** may be physically or **electronically transferred to** an acquirer, or successor or assignee as part of any merger, acquisition, debt financing, **sale of assets, or similar transaction**, as well as in the event of an insolvency, bankruptcy, or **receivership in which information is transferred to one or more third parties as one of our business assets**, for the sole purpose of continuing the **operation of the Service**, and only if the **recipient** of the Client Data commits to a Privacy Policy that has **terms** substantially **consistent** with this Privacy Policy

> You may **decline to share** certain Personal Data with us, in which case we may **not be able to provide to you** some of the features and **functionality** of the **Service**.

> The **Service** may **contain features or links to web sites and services provided by third parties**. Any **information you provide on third-party sites or services** is **provided directly to the operators** of such services and is **subject to those operators’ policies**, if any, governing privacy and security, even if accessed through the Service. We are **not responsible for the content or privacy and security practices and policies of third-party sites or services** to which **links or access are provided through the Service**.

> **Interest based advertising** is the **collection of data from different sources and across different platforms in order to predict an individual’s preferences or interest** and to deliver to that individual, or his/her computer, smart phone or tablet, advertising based on his/her **assumed preference or interest inferred from the collection of data pertaining to that individual** or others who may have a similar profile or similar interests.

> **We work with** a **variety of third parties** to **attempt to understand the profiles of** the **individuals** who are most likely to be interested in the Crowdin products or services so that we can send them promotional emails, or serve our advertisements to them **on the** **websites and mobile apps of other entities**.

> These third parties include: (i) **advertising networks**, which **collect information about** a **person’s interests** when that person **views or interacts with one of their advertisements**; (ii) **attribution partners**, which **measure the effectiveness of** certain **advertisements**; **and** (iii) **business partners**, which **collect information when a person views** or interacts with **one of their advertisements**.

> In **collaboration with** these **third parties**, we **collect information about** our customers, prospects and other **individuals over time and across different platforms** **when** they **use these platforms** or interact with them. **Individuals** may **submit information directly on our Sites or on platforms run by third parties**, **or by interacting** with us, our **advertisements**, or **emails** they receive **from us** **or** from **third parties**. **We** may **use special tools** that are commonly used **for this purpose**, such as **cookies, pixels**, and **similar technologies**. **We** may **have access to databases** of **information collected by** our **business partners**.

> The **information** we or **third party collect** enables us to **learn** what **purchases the person made**, what **ads or content the person sees**, on **which ads or links the person clicks**, and **other actions that the person takes** on our **Sites,** or **in response to** our **emails**, or when **visiting or using third parties’ platforms**.

> We, or **the third parties with which we work, use the information collected** as described above **to understand the various activities and behaviors of** our customers, **Site visitors and others**. We, or **these third parties**, do this **for many reasons**, including: to **recognize new or past visitors to our Sites**; to **present more personalized content**; to **provide** more useful and relevant **ads** - for example, **if we know what ads you are shown** we can try **not to show you the same ones repeatedly**; to **identify visitors across devices, sales channels, third party websites and Sites**, or to **display or send personalized or targeted ads** and **other custom content** that is more **focused on a person’s** perceived **interest in products or services similar to those** that **we offer**.

> Our **interest-based ads** may be **served to you in emails** or on **third-party platforms**. We may **serve these ads about our products or services** or send **commercial communications directly** ourselves or **through these third parties**. Crowdin does **not provide any personal information to the third party sites** that **display interest-based ads promoting Crowdin**.

(***Just have to interject here to differentiate between personal information and personally identifiable information, read on)

> We allow **third parties** with which we **have a separate agreement to use cookies** and **other technologies** to **collect information about your use of the Site**. These third parties include (i) **business partners, which collect information when you view or interact with one of their advertisements on the Site**; and (ii) **advertising networks**, which **collect information about your interests** when you view or interact with one of their **advertisements**.

> The **information gathered by** these **third parties** is used to **make predictions about your interests or preferences** so that they can **display advertisements or promotional material** on this Site and on **other sites** across the Internet **tailored to your** apparent **interests**.

> The **business partners** and **advertising networks** that **serve interest-based advertisements** on **the Services** have limited **access to** a small amount of **information about your profile and your device**, which is necessary **to serve you advertisements** that are **tailored to you**r apparent **interests**. It is possible that **they** may **reuse this** small amount of **information on other sites or services**.

(Bingo***)

> We do not share with these third parties any information that would readily identify you (such as email address); however, **these third parties** may have **access** to **information about you**r device (such as **IP or MAC address**). We do **no**t have access to, or **control over**, the **technologies** that these **third parties** may **use to collect information about you**r **interests**, and the **information practices of these third parties** are **not covered by this Privacy Notice**. **Other than** as discussed in **this document,** we have **no control over these third parties**.

(Interjecting again, e-mail address is identifiable, but IP and MAC isn't? Be the judge of this information.)

> **functionality cookies** - These cookies allow the website to **remember choices you make** (such as your **user name, language** or the **region** you are in) and provide enhanced, **more personal features**. For instance, **a website** may be able to provide you with local weather reports or traffic news by **storing in a cookie the region**
> in which **you are** currently **located**. These **cookies** can also be used to **remember changes** you have made to **text size, fonts** and **other parts of web pages that you can customise**. They may also be used to provide services you have asked for such as watching a video or commenting on a blog. The information these cookies collect **may be** anonymised and they **cannot track your browsing activity on other websites.**

(I'll let the reader be the judge of whether this in fact this grants a unique fingerprint that can track you across websites)

> **behaviourally targeted advertising cookies** - These cookies are used to deliver **adverts** more relevant to **you** and **your interests**. They are also used to limit the number of times you see an advertisement as well as help **measure the effectiveness of the advertising campaigns**. They are usually placed by advertising networks **with the website operator’s permission**. They **remember that you have visited a website** and this **information** is **shared with other organisations** such as **advertisers**. Quite often **targeting or advertising cookies will be linked to site functionality provided by the other organisation**.

> **Protecting the privacy of young children is especially important.** Our Service is not directed to children under the age of 18, and we do not knowingly collect Personal Data from children under the age of 18 without obtaining parental consent. **If you are under 18 years of age, then please do not use or access the Service at any time or in any manner.** If we learn that Personal Data has been collected on the Service from persons under 18 years of age and without verifiable parental consent, then we will take the appropriate steps to delete this information.

(Remember the part about good-faith earlier?)

> **The Service is not intended to be used** by minors, and is not intended **to** be used to **post content** to share publicly or **with friends**. To the extent that a minor has posted such content on the Service, the minor has the right to have this content deleted or removed using the deletion or removal options detailed in this Privacy Policy. If you have any question regarding this topic, please contact us as indicated in the “How to Contact Us” section. Please be aware that, **although we offer** this **deletion** capability**,** the removal of content may **not ensure** complete or comprehensive **removal of that content or information**.

> Although we may allow **you** to **adjust your privacy settings** to limit access to certain Personal Data, please be aware that no security measures are perfect or impenetrable. We are **not responsible for circumvention of any privacy settings** or security measures **on the Service**. Additionally, we **cannot control the actions of other users** with whom **you** may choose to **share** your **information**. Further, even **after information** posted on the Service **is removed**, **caching and archiving services** may have **saved that information**, and **other users or third parties** may have **copied or stored the information** available on the Service.

> **We** may **transfer, process and store Personal Data** we collect through the Services **in centralized databases** and with service providers **located in the U.S.** The U.S. may **not** have **the same data protection framework as the country from which you may be using** the Services.

(Take heed of why this needs to be pointed out. Do any particular US laws come to mind? Bingo.)

> The **Service** is **hosted in the United States** or Germany. **Regardless of the database** being **hosted in the EU**, if you choose to **use the Service from** the European Union or other **regions of the world with laws governing data collection** and use **that** may **differ from U.S. law, then** please note that **you** may be **transfer**ring your **Client Data and Personal Data** outside of those regions **to the United States for storage and processing** by our **service providers** listed in the Crowdin Terms of Service.

(But nobody is on that list, right?)

> **Crowdin** is **largely unaware of** what **Client Data** is actually being **stored or made available by a Client or User to the Service** and does not directly access such Client Data except as authorized by the Client, or as necessary to provide Services to the Client and its Users.

(It is almost as if the EU GDPR has provisions for this)

> Because Crowdin does not collect or determine the use of any Personal Data contained in the Client Data and because it does not determine the purposes for which such Personal Data is collected, the means of collecting such Personal Data, or the uses of such Personal Data, Crowdin is not acting in the capacity of data controller in terms of the European Union’s General Data Protection Regulation (Regulation (EU) 2016/679, hereinafter “GDPR”) and does not have the associated responsibilities under the GDPR. Crowdin should be considered only as a processor on behalf of its Clients and Users as to any Client Data containing Personal Data that is subject to the requirements of the GDPR. Except as provided in this Privacy Policy, Crowdin does not independently cause Client Data containing Personal Data stored in connection with the Services to be transferred or otherwise made available to third parties, except to third party subcontractors who may process such data on behalf of Crowdin in connection with Crowdin’s provision of Services to Clients. Such actions are performed or authorized only by the applicable Client or User.

(And it is almost like Crowdin has noticed)

> The Client or the User is the data controller under the Regulation for any Client Data containing Personal Data, meaning that such party controls the manner such Personal Data is collected and used as well as the determination of the purposes and means of the processing of such Personal Data.

> **Crowdin is not responsible for the content of the Personal Data** contained in the Client Data or **other information stored on its servers (or its subcontractors’ servers**) at the discretion of the Client or User nor is Crowdin responsible for the manner in which the Client or User collects, handles disclosure, **distributes** or otherwise **process**es such **information**.

> Please **revisit** this **page periodically to stay aware of any changes to this Policy**, which **we** may **update from time to time**. If we modify the Policy, **we** will make it available through the Service, and **indicate the date of the latest revision**, and will comply with applicable law. Your **continued use of the Service after** the revised Policy has become effective **indicates** that **you have read, understood and agreed** to the current version of the Policy.

(What a complete spaghetti cluster of vagueness, and outright scandalous terms and practices. Notice the pattern of _always_ mixing legitimate concerns with those that are not, the pattern of how they are referenced, and how connected parts are hidden in different categories, under headers that seem like they concern other concerns entirely. Look how the content of sentences is padded to seem less alarming, and how fill like "may", etc. just happen to appear exactly everywhere the alarming action is carried out.)

There is also the https://support.crowdin.com/cookies/ and https://downloads.crowdin.com/docs/DPA-singed.pdf
</details>

TL;DR You can't use the service without accepting, that your info is sold according to ability, for the purpose personal tracking, etc., you leak info to the NSA en-masse, all of the above happens live, and it doesn't matter if info is removed afterwards, and if it did, it is not promised that it will be gone, and the GDPR supposedly can't help you, because Crowdin doesn't know what goes on on their platform (they say), and doesn't think so (good luck with that), and you are supposed to be knowledgeable about both all of the above, and continually know what is in this document, which can change at any time, based on a date timestamp at the end.)

**Comment by asmeurer** [comment]
I disagree that crowdin is sublicensing the translations. A person contributing a translation is licensing it under the license for this repo, as well as under the crowdin terms that they agree to when signing up. 

I don't particularly care if Crowdin uses the translations for its own ML or whatever. Anyone could take the translations from the website itself or from the sources in this repo and do that anyway. It makes sense that they would explicitly require permissions via their ToS to do this so they don't have to worry about the notices part of the BSD license. As long as Crowdin allows the translations made on its site for the SymPy website to be contributed to the SymPy website under the respective open source license, I don't see any issues.

> In any event the term "Crowdin for Open Source" "We support the open source community" libre software package turns out to be only for "non-profit" projects, meaning no donations, no nothing. 

The word "non-profit" doesn't appear anywhere in the [crowdin terms of service](https://support.crowdin.com/terms/). On [this page](https://crowdin.com/page/open-source-project-setup-request), it only says "you do not have any commercial products related to the open-source project you are requesting a license for," which only means that the translations for the free "open source" tier would only be used for an open source project (which would be the case for the SymPy website). 

> TL;DR You can't use the service without accepting, that your info is sold according to ability, for the purpose personal tracking, etc., you leak info to the NSA en-masse, all of the above happens live, and it doesn't matter if info is removed afterwards, and if it did, it is not promised that it will be gone, and the GDPR supposedly can't help you, because Crowdin doesn't know what goes on on their platform (they say), and doesn't think so (good luck with that), and you are supposed to be knowledgeable about both all of the above, and continually know what is in this document, which can change at any time, based on a date timestamp at the end.)

What info would crowdin be collecting that isn't already public?

**Comment by comradekingu** [comment]
@asmeurer What alternative is there to using Crowdin if you want to contribute then? It snatches everything that enters the codebase on conditions of incredulity, subject to a court far outside the realms of the common translator.

I don't think translators really know what they are contributing to in that regard, and I can at-least attest to my motivation being that of only helping libre software projects. Then I actually read the terms and conditions/EULA. No you can't just lift the current website, as its basis constitutes a work, which is subject to copyright. Being a copyright holder depending on license chosen, (of which there is none now) means contributing.

However issueless you may deem the practice of shifting the axis of for profit motives, I think we can agree it is not optimal in its outcome. It does bring into question what alternatives could and do exist.

It says "If you're building awesome non-profit projects that could use the power of Crowdin, we're happy to help." and even interpreted in the meaning of related products, it means no commercialization. No T-shirts, no paid support, no support tiers. Transifex is already big enough to hold projects to the candle with those type of terms, and they do. Question why that (subject to change) text is even in the document at this point. As opposed to nothing, it serves a purpose. Look at the alternative endeavour, and you find no such clause. I see no issue in avoiding that pitfall. Using Crowdin means you can't ever fork it. If there is no issue moving platforms, sooner rather than later seems better.

If you click the "details" blurb above you will see Crowdin collects just about everything, and employs some spurious definitions and language to support it, in spaghetti fashion. Or you can load the site and plainly see.
From what I remember there is a list somewhere, and of course every single interaction is subject to respective terms and conditions. In the aggregate you have a valuable dataset that is hard enough to get even the basic of insight into. Again, the alternative is there and a much less complex proposal to similarly not take issue with in the current and foreseeable future.
